prov-api/routes/model_run/model_run.py

```diff
@@ -236,3 +238,82 @@ async def link_to_study(
         study_id=study_id,
         session_id=session_id
     )
+
+@router.post("/update", response_model=PostUpdateModelRunResponse, operation_id="update_model_run", include_in_schema=True)
+async def update_model_run(
+    payload: PostUpdateModelRunInput,
+    roles: ProtectedRole = Depends(read_write_user_protected_role_dependency),
+    config: Config = Depends(get_settings)
+) -> PostUpdateModelRunResponse:
+    # no proxy - user direct
+    request_style = RequestStyle(
+        service_account=None, user_direct=roles.user)
+
+    # validate first
+    valid, error_message = await validate_model_run_record(
+        record=payload.record,
+        request_style=request_style,
+        config=config,
+    )
+    if not valid:
+        assert error_message
+        raise HTTPException(
+            status_code=400,
+            detail=f"Failed to validate an entity's ID in the record, error: {error_message}."
+        )
+
+    res = await submit_model_run_update_job(
+        username=roles.user.username,
+        payload=ProvLodgeUpdatePayload(
+            model_run_record_id=payload.model_run_id,
+            updated_record=payload.record,
+            reason=payload.reason,
+            revalidate=True
+        ),
+        config=config)
+
+    return PostUpdateModelRunResponse(session_id=res)
+
+#
+#
+# class DeleteGraph(BaseModel):
+#    record_id: str
+#
+#
+# @router.post("/delete_graph", operation_id="delete_graph", include_in_schema=False)
+# async def delete_graph(
+#    delete: DeleteGraph,
+#    # admin only for this endpoint
+#    roles: ProtectedRole = Depends(admin_user_protected_role_dependency),
```

Comment:
need to keep/remove commented code?

Reply:
Want to keep this as it will be a basis for future delete endpoint.